waiter/src/waiter/process_request.clj

```diff
@@ -189,51 +188,31 @@
     (-> (ex-info message (assoc metrics-map :status status) error)
         (utils/exception->response request))))
 
-(defn http-method-fn
-  "Retrieves the qbits.jet.client.http client function that corresponds to the http method."
-  [request-method]
-  (let [default-http-method http/post
-        http-method-helper (fn http-method-helper [http-method]
-                             (fn inner-http-method-helper [client url request-map]
-                               (http/request client (into {:method http-method :url url} request-map))))]
-    (case request-method
-      :copy (http-method-helper :copy)
-      :delete http/delete
-      :get http/get
-      :head http/head
-      :move (http-method-helper :move)
-      :patch (http-method-helper :patch)
-      :post http/post
-      :put http/put
-      :trace http/trace
-      default-http-method)))
-
-(defn request->http-method-fn
-  "Retrieves the qbits.jet.client.http client function that corresponds to the http method used in the request."
-  [{:keys [request-method]}]
-  (http-method-fn request-method))
-
-(defn make-http-request
+(defn- make-http-request
   "Makes an asynchronous request to the endpoint using Basic authentication."
-  [http-client make-basic-auth-fn request-method endpoint headers body app-password {:keys [username principal]}
-   idle-timeout output-buffer-size]
+  [^HttpClient http-client make-basic-auth-fn request-method endpoint query-string headers body app-password
+   {:keys [username principal]} idle-timeout output-buffer-size]
   (let [auth (make-basic-auth-fn endpoint "waiter" app-password)
         headers (headers/assoc-auth-headers headers username principal)
-        http-method (http-method-fn request-method)]
-    (http-method
-      http-client endpoint
+        query-params (when-not (str/blank? query-string)
+                       (ring-codec/form-decode query-string))]
```

Comment:
`form-decode` does not work for us as it ends up parsing the query string wrong:

```
(ring-codec/form-decode "q=~`!@$%^&*()_-+={}[]|:;'<>,.?")
=> {"q" nil, "*()_- " "{}[]|:;'<>,.?"}
```

How do you guys feel about reporting `400 Bad Request`as the query string has invalid syntax?

Reply:
I'm thinking that simply catching the URISyntaxException and returning a 400 is looking like the best option.